packages/api/tests/endpoints/async-operations/test-create-async-operations.js

```diff
@@ -0,0 +1,387 @@
+    .set('Authorization', 'Bearer ThisIsAnInvalidAuthorizationToken')
+    .expect(401);
+
+  assertions.isInvalidAccessTokenResponse(t, response);
+  await t.throwsAsync(
+    () => t.context.asyncOperationModel.get({ id: asyncOperation.id }),
+    { instanceOf: RecordDoesNotExist }
+  );
+});
+
+test('POST creates a new async operation in all data stores', async (t) => {
+  const { asyncOperationPgModel, jwtAuthToken } = t.context;
+  const asyncOperation = fakeAsyncOperationFactory({
+    output: JSON.stringify({ age: 59 }),
+  });
+
+  const pgAsyncOperation = await translateApiAsyncOperationToPostgresAsyncOperation(asyncOperation);
+
+  const omitList = ['created_at', 'updated_at', 'cumulus_id', 'output'];
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .send(asyncOperation)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  const { message, record } = response.body;
+  const pgRecord = await asyncOperationPgModel.search(
+    t.context.testKnex,
+    { id: asyncOperation.id }
+  );
+
+  t.is(message, 'Record saved');
+  t.is(record.id, asyncOperation.id);
+  t.is(pgRecord.length, 1);
+
+  const [asyncOperationPgRecord] = pgRecord;
+
+  t.deepEqual(
+    omit(asyncOperationPgRecord, omitList),
+    omit(pgAsyncOperation, omitList)
+  );
+  t.deepEqual(asyncOperationPgRecord.output, pgAsyncOperation.output);
+
+  const esRecord = await t.context.esAsyncOperationsClient.get(
+    asyncOperation.id
+  );
+  t.like(esRecord, record);
+});
+
+test('POST creates a new async operation in DynamoDB and PG with correct timestamps', async (t) => {
+  const { asyncOperationModel, asyncOperationPgModel, jwtAuthToken } = t.context;
+  const asyncOperation = fakeAsyncOperationFactory({
+    output: JSON.stringify({ age: 59 }),
+  });
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .send(asyncOperation)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+  const { record } = response.body;
+
+  const fetchedRecord = await asyncOperationModel.get({
+    id: asyncOperation.id,
+  });
+
+  const pgRecord = await asyncOperationPgModel.search(
+    t.context.testKnex,
+    { id: asyncOperation.id }
+  );
+
+  const [asyncOperationPgRecord] = pgRecord;
+
+  t.true(fetchedRecord.createdAt > asyncOperation.createdAt);
+  t.true(fetchedRecord.updatedAt > asyncOperation.updatedAt);
+
+  const esRecord = await t.context.esAsyncOperationsClient.get(asyncOperation.id);
+
+  t.is(asyncOperationPgRecord.created_at.getTime(), record.createdAt);
+  t.is(asyncOperationPgRecord.updated_at.getTime(), record.updatedAt);
+  t.is(asyncOperationPgRecord.created_at.getTime(), esRecord.createdAt);
+  t.is(asyncOperationPgRecord.updated_at.getTime(), esRecord.updatedAt);
+});
+
+test('POST returns a 409 error if the async operation already exists in DynamoDB', async (t) => {
```

Comment:
I think this test should get updated to be checking if the record already exists in PostgreSQL in line with the change to the logic

Reply:
[`d2c9be3` (#2294)](https://github.com/nasa/cumulus/pull/2294/commits/d2c9be37d0bae4433225d328b530cd9be57da96d)